### PR TITLE
[NDMII-2518] Add ping configuration to template files

### PR DESCRIPTION
**Pull request by ken-schneider**
### What does this PR do?
Adds documentation to the config template files for NDM ping support

### Motivation
Document the changes added in this PR: https://github.com/DataDog/datadog-agent/pull/21939

### CODE REVIEW

**Review by jmw51798 — APPROVED** [review]

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -895,6 +895,33 @@ api_key:
   #
   # use_device_id_as_hostname: true
 
+  ## @param ping - custom object - optional
+  ## Configure ICMP pings for all hosts in SNMP autodiscovery
+  ## Each device will be pinged with these settings each time the SNMP
+  ## check is run.
+  ##
+  ## By default, we try to use an unprivileged UDP socket to send ICMP
+  ## pings, but some linux systems require using a raw socket.
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:904]:
```suggestion
  ## pings, but some Linux systems require using a raw socket.
```

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -895,6 +895,33 @@ api_key:
...
+  ## See details at the link below:
+  ## https://sturmflut.github.io/linux/ubuntu/2015/01/17/unprivileged-icmp-sockets-on-linux/
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:906]: Can we have this info in a DDOG-owned space? Seems a bit odd to link to a personal github page for info.

File: pkg/config/config_template.yaml [hunk 3]
```diff
@@ -895,6 +895,33 @@ api_key:
...
+  ##
+  ## If `linux.use_raw_socket` is set, you must enable the ping module
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:908]:
```suggestion
  ## If `linux.use_raw_socket` is set, you must enable the `ping` module
```

File: pkg/config/config_template.yaml [hunk 4]
```diff
@@ -895,6 +895,33 @@ api_key:
...
+  ## of the system-probe for elevated privileges. See
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:909]:
```suggestion
  ## of system-probe for elevated privileges. See
```

File: pkg/config/config_template.yaml [hunk 5]
```diff
@@ -1073,6 +1100,33 @@ api_key:
...
+    ## If `linux.use_raw_socket` is set, you must enable the ping module
+    ## of the system-probe for elevated privileges. See
+    ## system-probe.yaml.example for details.
+    #
+    # ping:
+    #   enabled: true             # disabled by default
+    #   timeout: 3000             # timeout in milliseconds
+    #   count: 2                  # number of ping packets to send per check run
+    #   interval: 10              # time between sending pings (up to `count` packets) in milliseconds
+    #   linux:                    # linux specific configuration
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:1122]:
```suggestion
    #   linux:                    # linux-specific configuration
```

File: pkg/config/config_template.yaml [hunk 6]
```diff
@@ -1073,6 +1100,33 @@ api_key:
...
+    #     use_raw_socket: true    # send pings in a privileged fashion using a raw socket
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:1123]: Should this part be in `{{- if .PingModule }}` block?

**Review by sgnn7 — COMMENTED** [review]
@ken-schneider Left some comments. Can you also link the PR that added the backing code of this change to the description for easy tracking?

File: pkg/config/config_template.yaml (hunk 6 above)
Comment (ken-schneider) [reply pkg/config/config_template.yaml:1123]: This part should not, there's two pieces potentially that a user would need to configure. The regular agent, and system-probe. The `PingModule` piece is configured to make sure that the system-probe config only shows up in the `system-probe.yaml.example`. The section highlighted here is part of the regular agent configuration.

File: pkg/config/config_template.yaml (hunk 2 above)
Comment (ken-schneider) [reply pkg/config/config_template.yaml:906]: That's a good idea. I think for the sake of getting this in before the freeze I'll just remove the link for now. I'll try to get a page together and link to it before the freeze and open a separate PR to add it.

**Comment by ken-schneider** [comment]
Hey @sgnn7, here's a link to the PR: https://github.com/DataDog/datadog-agent/pull/21939

File: pkg/config/config_template.yaml [hunk 7]
```diff
@@ -895,6 +895,31 @@ api_key:
   #
   # use_device_id_as_hostname: true
 
+  ## @param ping - custom object - optional
+  ## Configure ICMP pings for all hosts in SNMP autodiscovery
+  ## Each device will be pinged with these settings each time the SNMP
```
Comment (drichards-87) [inline pkg/config/config_template.yaml:900]:
```suggestion
  ## Devices will be pinged with these settings each time the SNMP
```

File: pkg/config/config_template.yaml [hunk 8]
```diff
@@ -895,6 +895,31 @@ api_key:
...
+  ## check is run.
+  ##
+  ## By default, we try to use an unprivileged UDP socket to send ICMP
```
Comment (drichards-87) [inline pkg/config/config_template.yaml:903]:
```suggestion
  ## By default, Datadog tries to use an unprivileged UDP socket to send ICMP
```

File: pkg/config/config_template.yaml [hunk 9]
```diff
@@ -895,6 +895,31 @@ api_key:
...
+  #   timeout: 3000             # timeout in milliseconds
+  #   count: 2                  # number of ping packets to send per check run
+  #   interval: 10              # time between sending pings (up to `count` packets) in milliseconds
+  #   linux:                    # linux specific configuration
+  #     use_raw_socket: true    # send pings in a privileged fashion using a raw socket
+  #                             # this may be required if your system doesn't support
+  #                             # sending pings in an unprivileged fashion (using a UDP socket)
+  #                             # If `use_raw_socket` is set to true, you MUST also enable the
+  #                             # system-probe which has elevated privileges, see system-probe.yaml.example
+  #                             # to enable it.
```
Comment (drichards-87) [inline pkg/config/config_template.yaml:921]:
```suggestion
  #                             # system-probe which has elevated privileges. To enable it, see system-probe.yaml.example.
  #                             
```

File: pkg/config/config_template.yaml [hunk 10]
```diff
@@ -1073,6 +1098,31 @@ api_key:
     #         - "monitored:true"
     #         - "customKey:customValue"
 
+    ## @param ping - custom object - optional
+    ## Configure ICMP pings for all hosts in SNMP autodiscovery
+    ## Each device will be pinged with these settings each time the SNMP
```
Comment (drichards-87) [inline pkg/config/config_template.yaml:1103]:
```suggestion
    ## Devices will be pinged with these settings each time the SNMP
```

File: pkg/config/config_template.yaml [hunk 11]
```diff
@@ -1073,6 +1098,31 @@ api_key:
...
+    ## check is run.
+    ##
+    ## By default, we try to use an unprivileged UDP socket to send ICMP
```
Comment (drichards-87) [inline pkg/config/config_template.yaml:1106]:
```suggestion
    ## By default, Datadog tries to use an unprivileged UDP socket to send ICMP
```

**Review by drichards-87 — APPROVED** [review]
Left a couple of suggestion from Docs and approved the PR.

**Review by sgnn7 — APPROVED** [review]

**Pull request merged**